### PR TITLE
test: add tests for `TbdexQuoteNotifier`

### LINKED ISSUE #293: Create tests for `TbdexQuoteNotifier`
- using the existing test suite as an example, create tests to stress-test `TbdexQuoteNotifier`
- the `TbdexQuoteNotifier` widget is responsible for:
  - polling a specific PFI for a specific exchange id and looking for a tbDEX `Quote` message with exponential backoff retries

### PR DESCRIPTION
**Pull request by mohitrajsinha**
This PR introduces several tests for `TbdexQuoteNotifier`.
closes #293

### CODE REVIEW

File: test/features/tbdex/tbdex_quote_notifier_test.dart [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  setUp(() {
+    mockTbdexService = MockTbdexService();
+    mockTbdexQuoteNotifier = MockTbdexQuoteNotifier();
+
+    when(() => mockTbdexService.getExchange(any(), any(), any()))
+        .thenAnswer((_) async => TestData.getExchange());
+  });
+
+  group('TbdexQuoteNotifier Stress Tests', () {
+    test('High-frequency polling with exponential backoff', () async {
```
Comment (ethan-tbd) [inline test/features/tbdex/tbdex_quote_notifier_test.dart:26]: same as https://github.com/TBD54566975/didpay/pull/312#discussion_r1797257937

File: test/features/tbdex/tbdex_quote_notifier_test.dart [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  setUp(() {
+    mockTbdexService = MockTbdexService();
+    mockTbdexQuoteNotifier = MockTbdexQuoteNotifier();
+
+    when(() => mockTbdexService.getExchange(any(), any(), any()))
+        .thenAnswer((_) async => TestData.getExchange());
+  });
+
+  group('TbdexQuoteNotifier Stress Tests', () {
```
Comment (ethan-tbd) [inline test/features/tbdex/tbdex_quote_notifier_test.dart:25]: let's rename this group to just `TbdexQuoteNotifier`

**Comment by ethan-tbd** [comment]
take a look at the tests for `AccountBalanceNotifier` and `TransactionNotifier` to see how to better structure your test cases :)

**Comment by mohitrajsinha** [comment]
Hey @ethan-tbd Check out the latest changes.

File: test/features/tbdex/tbdex_quote_notifier_test.dart [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    test('should handle network latency', () async {
+      when(() => mockTbdexQuoteNotifier.startPolling(any(), any()))
+          .thenAnswer((_) async {
+        await Future.delayed(const Duration(seconds: 5));
+        return TestData.getQuote();
+      });
+
+      await mockTbdexQuoteNotifier.startPolling('pfiId', 'exchangeId');
+
+      verify(() => mockTbdexQuoteNotifier.startPolling(any(), any())).called(1);
```
Comment (ethan-tbd) [inline test/features/tbdex/tbdex_quote_notifier_test.dart:56]: what's the reason we are using `any()` here in `startPolling()`?

File: test/features/tbdex/tbdex_quote_notifier_test.dart [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    test('should handle maximum retry attempts', () async {
+      when(() => mockTbdexQuoteNotifier.pollForQuote(any(), any()))
+          .thenThrow(Exception('Network error'));
+
+      await mockTbdexQuoteNotifier.startPolling('pfiId', 'exchangeId');
+
+      verify(() => mockTbdexQuoteNotifier.startPolling('pfiId', 'exchangeId'))
+          .called(1);
+    });
```
Comment (ethan-tbd) [inline test/features/tbdex/tbdex_quote_notifier_test.dart:45]: you are throwing an `Exception` here but the test case only checks to see if `startPolling()` has been called

File: test/features/tbdex/tbdex_quote_notifier_test.dart [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    test('should handle high-frequency polling with exponential backoff',
+        () async {
+      when(() => mockTbdexQuoteNotifier.pollForQuote(any(), any()))
+          .thenAnswer((_) async => TestData.getQuote());
+
+      await mockTbdexQuoteNotifier.startPolling('pfiId', 'exchangeId');
+
+      verify(() => mockTbdexQuoteNotifier.startPolling('pfiId', 'exchangeId'))
+          .called(1);
+    });
```
Comment (ethan-tbd) [inline test/features/tbdex/tbdex_quote_notifier_test.dart:35]: if you want to test the exponential backoff, try verifying how many times `getExchange()` is called

**Review by ethan-tbd — APPROVED** [review]
nice work!

**Pull request merged**